### PR TITLE
Common library bug fixes and code refactoring

### LINKED ISSUE #881: Quantity type is not supported yet for minValue and MaxValue extension
**Is your feature request related to a problem? Please describe.**
It gives error not implemented yet.

Implemented below unit test, but it had thrown quantity type is not implemented yet.

```
@Test
  @UiThreadTest
  fun displayValidationResult_validValue_shouldAssignErrorNull() {
    viewHolder.bind(
      QuestionnaireItemViewItem(
        Questionnaire.QuestionnaireItemComponent().apply {
          addExtension().apply {
            url = "http://hl7.org/fhir/StructureDefinition/minValue"
            setValue(Quantity(2.2))
          }
          addExtension().apply {
            url = "http://hl7.org/fhir/StructureDefinition/maxValue"
            setValue(Quantity(4.4))
          }
        },
        QuestionnaireResponse.QuestionnaireResponseItemComponent().apply {
          addAnswer(
            QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent().apply {
              value = Quantity(3.3)
            }
          )
        }
      ) {}
    )

    assertThat(viewHolder.itemView.findViewById<TextInputEditText>(R.id.textInputEditText).error).isNull()
  }

```

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by khyativyasargus**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #881 

**Description**
Added support to compare units even if canonical conversion fails & some code refactoring and bug fixes done in common lib

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: common/src/main/java/com/google/android/fhir/MoreTypes.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Google LLC
+ * Copyright 2022 Google LLC
```
Comment (jingtang10) [inline common/src/main/java/com/google/android/fhir/MoreTypes.kt:2]: can you update this file https://github.com/google/android-fhir/blob/master/license-header.txt to 2023?

File: common/src/main/java/com/google/android/fhir/UnitConverter.kt [hunk 2]
```diff
@@ -43,11 +42,11 @@ object UnitConverter {
    * Returns the canonical form of a UCUM Value.
    *
    * The canonical form is generated by normalizing [value] to UCUM base units, used to generate
-   * canonical matches on Quantity Search
+   * canonical matches on Quantity Search, if fails to generate then throws exception
```
Comment (jingtang10) [inline common/src/main/java/com/google/android/fhir/UnitConverter.kt:45]: use this instead https://kotlinlang.org/docs/kotlin-doc.html#throws-class-exception-class

File: datacapture/src/main/java/com/google/android/fhir/datacapture/validation/MinLengthValidator.kt [hunk 3]
```diff
@@ -41,9 +42,8 @@ internal object MinLengthValidator :
         (answer.value as PrimitiveType<*>).asStringValue().length <
           (extension.value as IntegerType).value
     },
-    messageGenerator = { extension: Extension, _: Context ->
-      ("The minimum number of characters that are permitted in the answer is: " +
-        extension.value.primitiveValue())
+    messageGenerator = { extension: Extension, context: Context ->
+      context.getString(R.string.min_length_validation_error_msg, extension.value.primitiveValue())
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/validation/MinLengthValidator.kt:46]: thanks for catching this!

File: common/src/main/java/com/google/android/fhir/MoreTypes.kt (hunk 1 above)
Comment (jingtang10) [reply common/src/main/java/com/google/android/fhir/MoreTypes.kt:2]: and rerun spotless

**Review by jingtang10 — APPROVED** [review]

**Review by jingtang10 — APPROVED** [review]

**Comment by omarismail94** [comment]
@khyativyasargus  this test is failing:
com.google.android.fhir.datacapture.validation.MinValueValidatorTest > should return invalid result when entered value is less than minValue for cqf calculated expression FAILED
    com.google.common.truth.ComparisonFailureWithFacts at MinValueValidatorTest.kt:149

**Pull request merged**